### PR TITLE
Tests for OnInvoke neo-cli

### PR DESCRIPTION
**Pull request by igormcoelho**
I guess we can separate a few methods to guarantee better testability directly on neo-cli project.

### CODE REVIEW

**Comment by igormcoelho** [comment]
@erikzhang @shargon this effectively creates a template which we can easily use for future testings.

File: tests/protocol.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "ProtocolConfiguration": {
```
Comment (erikzhang) [inline tests/protocol.json]: This file could be removed.

File: tests/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bin/
+obj/
```
Comment (erikzhang) [inline tests/.gitignore]: This file could be removed.

File: neo-cli/Shell/MainService.cs [hunk 3]
```diff
@@ -184,9 +184,9 @@ private bool OnDeployCommand(string[] args)
             return SignAndSendTx(tx);
         }
 
-        private bool OnInvokeCommand(string[] args)
+        internal protected void ParseParametersInvoke(string[] args, out UInt160 scriptHash, out ContractParameter[] parameters)
```
Comment (erikzhang) [inline neo-cli/Shell/MainService.cs]: Why we need this?

Comment (igormcoelho) [reply neo-cli/Shell/MainService.cs]: I separated OnInvokeCommand in two parts, to allow testing of this method (without getting to the last part of tx relay). This allowed to check that parameters were being parsed, and that transaction was being built correctly.
Still, I agree it's kind of ugly that we cannot fully test the complete thing (OnInvokeCommand). I think I just got a better idea.

**Comment by igormcoelho** [comment]
@erikzhang @shargon this is the complete template for tests, I just removed any other extra change.
I have some ideas for good testing here, but I guess we should start applying these on Neo 3.0 first (neo-cli 3.x).

File: .gitignore [hunk 4]
```diff
@@ -1,288 +0,0 @@
-## Ignore Visual Studio temporary files, build results, and
```
Comment (vncoelho) [inline .gitignore]: Can this .gitignore be really removed?

Comment (igormcoelho) [reply .gitignore]: did I remove something? I'll check.

Comment (igormcoelho) [reply .gitignore]: @vncoelho

**Pull request merged**